### DISCUSSION TITLE
Rewrite Bigdata in Rust

### DISCUSSION THREAD

**Discussion by giscus** [Xuanwo/blog#429] [Announcements]
# Rewrite Bigdata in Rust

An infrastructure engineer, focused on distributed storage system

https://xuanwo.io/2024/07-rewrite-bigdata-in-rust/

**Comment by drc-infinyon** [comment]
I cant see Fluvio in your list - https://github.com/infinyon/fluvio

**Reply by drc-infinyon** [reply]
Fluvio is an edge to core cloud native streaming engine built from the ground up in rust. Compiles to a single 37 Meg binary and deploys on ARM64 devices.

We just released the first public beta version of Stateful DataFlow.

Stateful DataFlow is a framework for building unbounded distributed stream processing based on wasm that runs on Fluvio streams.

We are going for a Lean alternative to Kafka + Flink with a user experience of Ruby on Rails.

BTW, Stateful DataFlow has integrations with Arrow, Polars, and the ability to use SQL for dataframes, and other wasm compatible programming languages to express business logic. And Fluvio has Rust, Python, and JS clients.

**Reply by Xuanwo** [reply]
Great, thanks for sharing. I'll add it to the list.

**Comment by a-agmon** [comment]
I think you can also add LanceDB - which is a vector db, with arrow based API,  that separate storage and compute - https://github.com/lancedb/lancedb
I wrote about it  [in this post](https://medium.com/towards-data-science/scale-up-your-rag-a-rust-powered-indexing-pipeline-with-lancedb-and-candle-cc681c6162e8)

**Reply by Xuanwo** [reply]
Great!

**Comment by gauravkumar37** [comment]
Can you please add https://slatedb.io/ https://github.com/slatedb/slatedb to the list?
Thanks

**Comment by kfehlhauer** [comment]
Can you please add https://github.com/delta-io/delta-rs to the list?

**Reply by Xuanwo** [reply]
Great! I will add them in https://github.com/rewrite-bigdata-in-rust/RBIR.

**Reply by kfehlhauer** [reply]
Thanks! This is also incubating.

https://github.com/delta-incubator/delta-kernel-rs

**Reply by Xuanwo** [reply]
> Thanks! This is also incubating.
> 
> [delta-incubator/delta-kernel-rs](https://github.com/delta-incubator/delta-kernel-rs)

Wow, that's new to me. Great!

**Comment by bjm1244** [comment]
There is a dask replacement, polars.
https://pola.rs/

I am currently interested in a data system composed of Rust, such as databend, polars, and would like to contribute to this.
